tajo-algebra/src/main/java/org/apache/tajo/algebra/DataTypeExpr.java

```diff
@@ -59,15 +68,34 @@ public String getTypeName() {
   }
 
   public boolean isPrimitiveType() {
-    return !this.isRecordType() && !isMapType();
+    return !isArrayType()&& !isRecordType() && !isMapType();
+  }
+
+  public boolean isArrayType() {
+    return arrayType != null;
   }
 
   public boolean isRecordType() {
-    return this.typeName.equals(Type.RECORD.name());
+    return recordType != null;
```

Comment:
Will this method be frequently called? It seems not, and I think it would be better to check the type name because we can figure out when the type is record but ```recordType``` is null.